server/planning/events/events_update_repetitions.py

```diff
@@ -8,235 +8,233 @@
+        post = {
+            "event": event[ID_FIELD],
+            "etag": event["_etag"],
+            "update_method": "single",
+            "pubstatus": event.get("pubstatus"),
+        }
+        await get_resource_service("events_post").post_async([post])
+
+
+async def delete_event(event: dict[str, Any], updated_rule: dict[str, Any]):
+    events_service = EventResourceModel.get_service()
+
+    if event.get("pubstatus", None) is not None or event_has_planning_items(event[ID_FIELD], "primary"):
+        await cancel_event(event, updated_rule)
+    else:
+        await events_service.delete_many(lookup={"_id": event[ID_FIELD]})
+        app = get_current_app().as_any()
+        app.on_deleted_item_events(event)
+
+
+def create_event(date, updates: dict[str, Any], original: dict[str, Any], time_delta):
+    # Create a copy of the metadata to use for the new event
+    new_event = deepcopy(original)
+    new_event.update(deepcopy(updates))
+
+    # Remove fields not required by new events
+    remove_fields(new_event, extra_fields=["reschedule_from", "pubstatus"])
+
+    new_event["state"] = WORKFLOW_STATE.DRAFT
+    for key in list(new_event.keys()):
+        if key.startswith("_") or key.startswith("lock_"):
+            new_event.pop(key)
+
+    # Set the new start and end dates, as well as the _id and guid fields
+    new_event["dates"]["start"] = date
+    new_event["dates"]["end"] = date + time_delta
+    new_event[ID_FIELD] = new_event["guid"] = generate_guid(type=GUID_NEWSML)
+    set_original_creator(new_event)
+    set_planning_schedule(new_event)
 
-    def _create_event(self, date, updates, original, time_delta):
-        # Create a copy of the metadata to use for the new event
-        new_event = deepcopy(original)
-        new_event.update(deepcopy(updates))
+    return new_event
 
-        # Remove fields not required by new events
-        EventsUpdateRepetitionsService.remove_fields(new_event, extra_fields=["reschedule_from", "pubstatus"])
 
-        new_event["state"] = WORKFLOW_STATE.DRAFT
-        for key in list(new_event.keys()):
-            if key.startswith("_") or key.startswith("lock_"):
-                new_event.pop(key)
+async def update_event(updated_rule: dict[str, Any], original: dict[str, Any]):
+    events_service = EventResourceModel.get_service()
+    events_history_service = EventsHistoryResourceModel.get_service()
 
-        # Set the new start and end dates, as well as the _id and guid fields
-        new_event["dates"]["start"] = date
-        new_event["dates"]["end"] = date + time_delta
-        new_event[ID_FIELD] = new_event["guid"] = generate_guid(type=GUID_NEWSML)
-        set_original_creator(new_event)
-        self.set_planning_schedule(new_event)
+    event_id = original[ID_FIELD]
+    updates = update_rules(original, updated_rule)
+    set_planning_schedule(updates)
+    await events_service.system_update(event_id, updates)
```

Comment:
And here too, use `system_update_async`

Reply:
@MarkLark86 in this function I am using EventAsyncService and not the old event_service that's now using AsyncBaseService. 

In instances I could make use of the new Pydantic models/async services I use them like here.